eng/pipelines/templates/stages/archetype-cpp-release.yml

```diff
@@ -214,4 +181,59 @@ stages:
                           PRBranchName: increment-package-version-${{ parameters.ServiceDirectory }}-$(Build.BuildId)
                           CommitMsg: "Increment package version after release of ${{ artifact.name }}"
                           PRTitle: "Increment version for ${{ parameters.ServiceDirectory }} releases"
-                          CloseAfterOpenForTesting: '${{ parameters.TestPipeline }}'
\ No newline at end of file
+                          CloseAfterOpenForTesting: '${{ parameters.TestPipeline }}'
+
+  - ${{if eq(variables['System.TeamProject'], 'internal') }}:
+    - stage: Integration
+      dependsOn: Build
+      condition: or(eq(variables['PublishNightlyVcpkg'], 'true'), eq(variables['Build.Reason'],'Schedule'))
+      jobs:
+        - job: PublishNightlyVcpkg
+          displayName: Publish to vcpkg nightly branch
+
+          pool:
+            name: azsdk-pool-mms-win-2019-general
+            vmImage: MMS2019
+
+          steps:
+            - checkout: self
+            - download: current
+              artifact: packages
+
+            - template: /eng/pipelines/templates/steps/generate-nightly-branch-name.yml
+
+            - template: /eng/pipelines/templates/steps/vcpkg-clone.yml
+              parameters:
+                RepoOwner: azure-sdk
+                PRBranchName: $(NightlyBranchName)
+
+            # Placing artifacts in the same branch in sequence reduces the
+            # potential for race conditions. If this job does fail to push
+            # because another job has already pushed to the same branch it
+            # can be re-run.
+            - ${{ each artifact in parameters.Artifacts }}:
+              - template: /eng/pipelines/templates/steps/vcpkg-publish.yml
+                parameters:
+                  ArtifactName: ${{ artifact.Name }}
+                  VcpkgPortName: ${{ artifact.VcpkgPortName }}
+                  DisplayNameExtension: ${{ artifact.Name }}
+                  TestRelease: true
+                  TestReleaseRef: $(Build.SourceVersion)
+
+            - pwsh: |
+                Write-Host "git reset --soft HEAD~${{ length(parameters.Artifacts) }}"
+                git reset --soft HEAD~${{ length(parameters.Artifacts) }}
```

Comment:
Is there a reason to the the reset here and not in the individual publish steps so it is a single reset of the commit that was made?

Reply:
Also why a soft reset instead of the default which is mixed? https://git-scm.com/docs/git-reset#Documentation/git-reset.txt---mixed

Reply:
Another reason it's better to have resets inside the loop: Some libraries don't publish to vcpkg yet. In those cases those packages do not count toward the `length(parameters.Artifacts)` and we would perform too many resets. 

Soft reset was used because it was helpful to the iterating I was doing at some point. It's removed now.